src/algebra/ring/basic.lean

```diff
@@ -202,10 +202,14 @@ end add_monoid_hom
 
 /-- Bundled semiring homomorphisms; use this for bundled ring homomorphisms too. -/
 structure ring_hom (α : Type*) (β : Type*) [semiring α] [semiring β]
-  extends monoid_hom α β, add_monoid_hom α β
+  extends monoid_hom α β, add_monoid_hom α β, monoid_with_zero_hom α β
 
 infixr ` →+* `:25 := ring_hom
 
+/-- Reinterpret a ring homomorphism `f : R →+* S` as a `monoid_with_zero_hom R S`.
+The `simp`-normal form is `(f : monoid_with_zero_hom R S)`. -/
+add_decl_doc ring_hom.to_monoid_with_zero_hom
```

Comment:
This doesn't seem to match with the other `coe`s between homs you added, is the reason backwards compatibility or something else?

Reply:
Yeah, I left the simp normal form unchanged for the other ones to save me work. I don't have a strong preference for which one should be simp normal - depends if you want to see `f.to_monoid_with_zero_hom.to_monoid_hom` or `\u\uf` in the goal state

Reply:
OK, that's fine. Could you document this in the code? (a short comment in both places pointing out the difference is enough)

Reply:
I've gone for just a comment in `group/hom.lean`. I suspect these coercions exist all over mathlib, likely with different preferred directions - the new comment explains why `hom` is done the way it is, but that conventions may vary across mathlib.